LEGO1/lego/legoomni/src/worlds/legoact2.cpp

```diff
@@ -90,14 +129,116 @@ MxResult LegoAct2::Create(MxDSAction& p_dsAction)
 	return result;
 }
 
-// STUB: LEGO1 0x10050040
+// FUNCTION: LEGO1 0x10050040
+// FUNCTION: BETA10 0x1003a976
 MxResult LegoAct2::Tickle()
 {
-	// TODO
-	return SUCCESS;
+	MxFloat distance;
+
+	if (!m_worldStarted) {
+		LegoWorld::Tickle();
+		return 0;
+	}
+
+	switch (m_unk0x10c4) {
+	case 0:
+		m_unk0x10c4 = 1;
+		break;
+	case 1:
+		((LegoPathActor*) m_unk0x10d8->GetEntity())->SetState(LegoPathActor::c_bit3);
+
+		switch (rand() % 3) {
+		case 0:
+			g_unk0x100f4474 = Act2mainScript::c_tns002br_RunAnim;
+			break;
+		case 1:
+			g_unk0x100f4474 = Act2mainScript::c_tns003br_RunAnim;
+			break;
+		case 2:
+			g_unk0x100f4474 = Act2mainScript::c_tns004br_RunAnim;
+			break;
+		}
+
+		FUN_10052560(g_unk0x100f4474, TRUE, TRUE, NULL, NULL, NULL);
+		m_unk0x10d0 = 0;
+		m_unk0x10c4 = 2;
+		break;
+	case 2:
+		if (g_unk0x100f4474) {
+			if (AnimationManager()->FUN_10064ee0(g_unk0x100f4474)) {
+				FUN_10015820(FALSE, LegoOmni::c_disableInput | LegoOmni::c_disable3d | LegoOmni::c_clearScreen);
+				g_unk0x100f4474 = 0;
+			}
+		}
+
+		m_unk0x10d0 += 50;
+		break;
+	case 3:
+		FUN_10015820(FALSE, LegoOmni::c_disableInput | LegoOmni::c_disable3d | LegoOmni::c_clearScreen);
+		m_unk0x10d0 = 0;
+		m_unk0x10c4 = 4;
+		FUN_10052560(Act2mainScript::c_tja009ni_RunAnim, TRUE, TRUE, NULL, NULL, NULL);
+
+		AnimationManager()->EnableCamAnims(TRUE);
+		AnimationManager()->FUN_1005f6d0(TRUE);
+		AnimationManager()->FUN_100604f0(&g_unk0x100f43f0[0], sizeOfArray(g_unk0x100f43f0));
```

Comment:
would just write `g_unk0x100f43f0` here instead of `&g_unk0x100f43f0[0]` (same as in the next line)

Reply:
Not sure why that didn't work before, but now it does - thanks!